frontend/src/components/Collection/components/CollectionDescription/index.tsx

```diff
@@ -0,0 +1,132 @@
+import { Intent } from "@blueprintjs/core";
+import React, { useCallback, useEffect, useRef, useState } from "react";
+import {
+  CollectionDescription as Description,
+  DescriptionText,
+  DESCRIPTION_LINE_HEIGHT_PX,
+  MAX_LINE_COUNT,
+} from "src/components/Collection/components/CollectionDescription/style";
+import { StyledPrimaryAnchorButton } from "src/components/common/Button/common/style";
+
+enum EllipsisMode {
+  "NONE" = "NONE",
+  "OFF" = "OFF",
+  "ON" = "ON",
+}
+
+interface Props {
+  description: string;
+}
+
+export default function CollectionDescription({
+  description,
+}: Props): JSX.Element {
+  const descriptionRef = useRef<HTMLParagraphElement>(null);
+  const [ellipsisMode, setEllipsisMode] = useState<EllipsisMode>(
+    EllipsisMode.NONE
+  );
+  const isEllipsis = ellipsisMode === EllipsisMode.ON;
+
+  /**
+   * Updates the state of ellipsis mode by toggling between "ON" and "OFF" values.
+   */
+  const onToggleMode = (): void => {
+    setEllipsisMode((currentMode) => {
+      if (currentMode === EllipsisMode.ON) {
+        return EllipsisMode.OFF;
+      }
+      return EllipsisMode.ON;
+    });
+  };
+
+  /**
+   * Updates ellipsis mode state.
+   */
+  const onUpdateMode = useCallback((): void => {
+    if (descriptionRef.current) {
+      const el = descriptionRef.current;
+      setEllipsisMode((currentMode) => getEllipsisMode(el, currentMode));
+    }
+  }, []);
+
+  useEffect(() => {
+    onUpdateMode(); /* Initialize ellipsis mode. */
+    window.addEventListener("resize", onUpdateMode);
```

Comment:
discussion: Is this a good use case for ResizeObserver 🙆‍♂️ ?

https://web.dev/resize-observer/

Sounds like there's perf upside!

Thank you!

Reply:
oooooo, @tihuan, that's awesome...